src/test/test_proj_data_info_subsets.cxx

```diff
@@ -370,84 +374,57 @@ generate_full_forward_projection(
     const shared_ptr<const ProjData>& template_sino_sptr,
     bool use_z_symmetries, bool use_other_symmetries)
 {
-  if (input_image_sptr->find_max() == 0) {
-    cerr << "error: Tests are run with redundant empty image" << endl;
-    everything_ok = false;
-  }
+  check (input_image_sptr->find_max() > 0, "forward projection test run with empty image");
 
   auto fwd_projector_sptr = construct_projector_pair(
     template_sino_sptr->get_proj_data_info_sptr(), input_image_sptr, use_z_symmetries, use_other_symmetries)
       ->get_forward_projector_sptr();
   
-  auto full_forward_projection = ProjDataInMemory(*template_sino_sptr);
+  ProjDataInMemory full_forward_projection(*template_sino_sptr);
   fwd_projector_sptr->set_input(*input_image_sptr);
   fwd_projector_sptr->forward_project(full_forward_projection);
 
-  if (full_forward_projection.get_viewgram(0, 0).find_max() == 0) {
-    cerr << "error: segment 0, view 0 of reference forward projection is empty!" << endl;
-    everything_ok = false;
-  }
+  check(full_forward_projection.get_viewgram(0, 0).find_max() > 0, "segment 0, view 0 of reference forward projection is empty");
 
   return full_forward_projection;
 }
 
 
-VoxelsOnCartesianGrid<float> TestProjDataInfoSubsets::
+shared_ptr<VoxelsOnCartesianGrid<float>> TestProjDataInfoSubsets::
 generate_full_back_projection(
     const shared_ptr<const ProjData>& input_sino_sptr,
     const shared_ptr<const VoxelsOnCartesianGrid<float> >& template_image_sptr,
     bool use_z_symmetries, bool use_other_symmetries)
 {
-  bool input_is_nonzero = false;
```

Comment:
Are you sure we don't want this? I included it originally to double check I'd generated test data correctly. But if the tests are run on empty input data they will always trivially pass.

Reply:
I know, but I guess we know now that you don't have a bug in the test... I want to reduce the number of loops over data as much as possible as that will change with TOF